pkg/route/api/v1/types.go

```diff
@@ -48,6 +48,13 @@ type RouteSpec struct {
 
 	// TLS provides the ability to configure certificates and termination for the route
 	TLS *TLSConfig `json:"tls,omitempty" description:"provides the ability to configure certificates and termination for the route"`
+
+	// RouterShard has information of a routing shard and is used to
+	// generate host names
+	Shard *RouterShard `json:"shard,omitempty" description:"provides shard information for the route"`
```

Comment:
For context here is the minimal set of changes that are targeted - https://github.com/openshift/origin/pull/4553.  Routes will contain status, status is 0..N, status will convey the actual host conveyed to the user.